File: apps/codemap-search/experiments/event-navigation-poc/language-corpus/fixtures/ruby/callbacks.rb
class Router
  def set_primary(cb)
    @primary = cb # store_primary
  end
  def set_secondary(cb)
    @secondary = cb # store_secondary
  end
  def fire_primary
    @primary.call # call_primary
  end
  def fire_secondary
    @secondary.call # call_secondary
  end
end
class Other
  def fire_primary
    @primary.call # call_other
  end
end

File: apps/codemap-search/experiments/event-navigation-poc/language-corpus/fixtures/dart/callbacks.dart
class Router {
  void Function() primary = () {};
  void Function() secondary = () {};
  void setPrimary(void Function() cb) { this.primary = cb; } // store_primary
  void setSecondary(void Function() cb) { this.secondary = cb; } // store_secondary
  void firePrimary() { primary.call(); } // call_primary
  void fireSecondary() { secondary.call(); } // call_secondary
}
class Other {
  void Function() primary = () {};
  void firePrimary() { primary.call(); } // call_other
}
